S2VX.Game/Story/Note/GameNote.cs

```diff
@@ -1,11 +1,127 @@
-﻿using osu.Framework.Input.Events;
+﻿using osu.Framework.Allocation;
+using osu.Framework.Input.Events;
+using osu.Framework.Screens;
+using osu.Framework.Utils;
+using osuTK.Graphics;
+using osuTK.Input;
+using S2VX.Game.Play;
 using System;
 
 namespace S2VX.Game.Story.Note {
     public class GameNote : S2VXNote {
-        protected override bool OnClick(ClickEvent e) {
-            Console.WriteLine("GameNote clicked");
+
+        [Resolved]
+        private S2VXScore Score { get; set; }
+
+        [Resolved]
+        private S2VXStory Story { get; set; }
+
+        [Resolved]
+        private ScreenStack ScreenStack { get; set; }
+
+        private int TimingError;
+
+        private void Delete() {
+            var playScreen = (PlayScreen)ScreenStack.CurrentScreen;
+            playScreen.PlayInfoBar.RecordHitError(TimingError);
+            Story.RemoveNote(this);
+        }
+
+        private void RecordMiss() {
+            var missThreshold = Story.Notes.MissThreshold;
+            TimingError = missThreshold;
+            Score.Value += missThreshold;
+            Delete();
+        }
+
+        private GameNote GetEarliestClickableNote() {
```

Comment:
As discussed, we should look in a better sorted list solution rather than run this every click.

Reply:
I sorted Notes, pull master for the latest changes.